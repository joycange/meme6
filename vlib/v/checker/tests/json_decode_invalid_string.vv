import json

struct TestStruct {
	host string
}

txt := '"{"host": "localhost"}'
json.decode(TestStruct, txt)!
json.decode(TestStruct, txt.trim_string_right('"'))!
json.decode(TestStruct, 'h{ost: "localhost"}')!
