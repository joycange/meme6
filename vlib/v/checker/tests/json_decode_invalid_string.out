vlib/v/checker/tests/json_decode_invalid_string.vv:7:8: error: json.decode: invalid json string
    5 | }
    6 | 
    7 | txt := '"{"host": "localhost"}'
      |        ~~~~~~~~~~~~~~~~~~~~~~~~
    8 | json.decode(TestStruct, txt)!
    9 | json.decode(TestStruct, txt.trim_string_right('"'))!
vlib/v/checker/tests/json_decode_invalid_string.vv:10:25: error: json.decode: invalid json string
    8 | json.decode(TestStruct, txt)!
    9 | json.decode(TestStruct, txt.trim_string_right('"'))!
   10 | json.decode(TestStruct, 'h{ost: "localhost"}')!
      |                         ~~~~~~~~~~~~~~~~~~~~~
